### ISSUE TITLE
Default constructors for GFs

### ISSUE THREAD

**Issue by galexv** [ALPSCore/ALPSCore#277]
Markus (@mwallerb) needs GFs to be default-constructible (e.g., to initialize them in an object constructor's body and catch any exceptions that might be thrown).

**Comment by galexv** [comment]
@shinaoka said:
> What if we make Meshes and GFs default-constructible (in C++)? What (bad) side effects
> do you expect?
> It may be less fail safe because users may attempt to use an uninitialized object.

Honestly, I dislike "uninitialized" objects; either user risks to get, silently, a garbage (or, at best, SIGSEGV) attempting to access such an object, or we have to introduce checks at every access whether the object is in a valid state.  

When it was first coded, there was no clear idea what could be an "empty" GF. :) Moreover, right now it is not possible to assign GFs with a different meshes; so, once a GF is created, its meshes cannot be changed.

As it seems to be a popular feature request, we will make GFs default-constructible. They can be assigned some default zero-sized meshes initially; an assignmentto a GF should also assign the new meshes, then.

**Comment by shinaoka** [comment]
Thank you ! Do you implement these features? Or, can I take this?

**Comment by egull** [comment]
Can you take it, and Alex can do code review?

**Comment by shinaoka** [comment]
OK. To avoid conflict, once the pull request on numerical mesh is merged into the master branch, I will make GFs default-constructible.

**Comment by galexv** [comment]
I believe this one is closed by PR #300 (thanks Hiroshi @shinaoka !)
